### DISCUSSION TITLE
Move fastlane changelogs into its own component

### DISCUSSION THREAD

**Discussion by LeoColman** [WeblateOrg/weblate#11523] [General] [answered]
This was already discussed in #7917 but I don't believe the answer is sufficient to the broader use case.

Even though we can
- Set the string to read-only
- Fastlane changelogs are lowest-priority shown to translators

I still have some problems I can't address:

- Read-only strings won't allow translators willing to translate changelogs to do it, reducing overall translatability
- Bundling both in the same component doesn't allow vision of true translation of the app -- Changelogs are not exactly part of the app and shouldn't count to the % translated of the language
- Changelog files (if left write-enabled) taint translation percentage of more-valuable app details

Currently I have an app that has "Translated %" going down because of new English-only changelogs. Some Spanish translators like to translate changelogs, but German ones won't do the same. The app has the same strings, but is "Better translated to Spanish".

----

## Technical Solution

I'm not sure what exactly to suggest, but the only thing I can think of is getting this screen:

![image](https://github.com/WeblateOrg/weblate/assets/1577251/0d0271df-5569-465b-a62c-51ed1101b8d9)

and adding an exclusion (i.e. "Exclude files matching fastlane/metadata/android/changelogs/*") filter.

This would allow me to create an "App Details" component with the exclusion and "Changelogs" with the other way around.

**Comment by nijel** [comment, marked as answer]
I think the best approach would be to support filtering based on key, what would make it a generic feature usable for all monolingual file formats. To keep consistency, it should behave like the existing Language filter. I think there is some issue for a feature like this, but I can't find it right now. If you are better at searching please post followup there, if not, please create a new issue.

**Reply by LeoColman** [reply]
I couldn't find an issue, so I guess we need a third person to try to search for it 😂 

https://github.com/WeblateOrg/weblate/issues/11562

Created. I'll mark this discussion as solved and await feedback on the issue :)

Thanks @nijel !
